WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/editor/WatchFaceConfigStateHolder.kt

```diff
@@ -0,0 +1,245 @@
+        val minuteHandStyle =
+            editorSession.userStyle[minuteHandLengthKey] as UserStyleSetting.DoubleRangeUserStyleSetting.DoubleRangeOption
+
+        Log.d(TAG, "/new values: $colorStyle, $ticksEnabledStyle, $minuteHandStyle")
+
+        val watchFacePreview = UserStylesAndPreview(
+            colorStyleId = colorStyle.id.toString(),
+            ticksEnabled = ticksEnabledStyle.value,
+            minuteHandLength = multiplyByMultipleForSlider(minuteHandStyle.value).toFloat(),
+            previewImage = bitmap
+        )
+
+        mutableUiState.value = EditWatchFaceUiState.Success(watchFacePreview)
+    }
+
+    fun setColorStyle(newColorStyleId: String) {
+        val userStyleSettingList = editorSession.userStyleSchema.userStyleSettings
+
+        // Loops over all UserStyleSettings (basically the keys in the map) to find the setting for
+        // the color style (which contains all the possible options for that style setting).
+        for (userStyleSetting in userStyleSettingList) {
+            if (userStyleSetting.id == UserStyleSetting.Id(COLOR_STYLE_SETTING)) {
+                val colorUserStyleSetting =
+                    userStyleSetting as UserStyleSetting.ListUserStyleSetting
+
+                // Loops over the UserStyleSetting.Option colors (all possible values for the key)
+                // to find the matching option, and if it exists, sets it as the color style.
+                for (colorOptions in colorUserStyleSetting.options) {
+                    if (colorOptions.id.toString() == newColorStyleId) {
+                        setUserStyleOption(colorStyleKey, colorOptions)
+                        return
+                    }
+                }
+            }
+        }
+    }
+
+    fun setDrawPips(enabled: Boolean) {
+        setUserStyleOption(
+            drawPipsKey,
+            UserStyleSetting.BooleanUserStyleSetting.BooleanOption.from(enabled)
+        )
+    }
+
+    fun setMinuteHandArmLength(newLengthRatio: Float) {
+        val newMinuteHandLengthRatio = newLengthRatio.toDouble() / MULTIPLE_FOR_SLIDER
+
+        setUserStyleOption(
+            minuteHandLengthKey,
+            UserStyleSetting.DoubleRangeUserStyleSetting.DoubleRangeOption(newMinuteHandLengthRatio)
+        )
+    }
+
+    // Saves User Style Option change back to the back to the EditorSession.
+    private fun setUserStyleOption(
+        userStyleSetting: UserStyleSetting,
+        userStyleOption: UserStyleSetting.Option
+    ) {
+        Log.d(WatchFaceConfigActivity.TAG, "setUserStyleOption()")
+        Log.d(WatchFaceConfigActivity.TAG, "\tuserStyleSetting: $userStyleSetting")
+        Log.d(WatchFaceConfigActivity.TAG, "\tuserStyleOption: $userStyleOption")
+
+        val mutableUserStyle = editorSession.userStyle.toMutableUserStyle()
+        mutableUserStyle[userStyleSetting] = userStyleOption
+        editorSession.userStyle = mutableUserStyle.toUserStyle()
+        updatesWatchFacePreview()
```

Comment:
There's a race here, since the initialization of `editorSession` occurs asynchronously. It's possible that a user updates an option before `editorSession` is initialized, which would cause a crash.

Reply:
Ah, I remembered that the buttons will only be enabled once the editor session is initialized. So we should be good here then.

Reply:
I added a note at the top of the class to call that out.